frontend/src/lib/components/QueryStateWrapper/queryStateWrapper.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+
+import { resolveClassNames } from "@lib/utils/resolveClassNames";
+import { QueryObserverResult } from "@tanstack/react-query";
+
+// Definition of error criteria for multiple queries
+// - SOME_QUERY: If at least one query has an error, the error component is displayed
+// - EVERY_QUERY: If all queries have an error, the error component is displayed
+export enum QueriesDisplayErrorCriteria {
+    SOME_QUERY = "some_query",
+    EVERY_QUERY = "every_query",
+}
+
+// Base state wrapper props
+export type QueryStateWrapperBaseProps = {
+    loadingComponent: React.ReactNode;
+    errorComponent: React.ReactNode;
+    className?: string;
+    style?: React.CSSProperties;
+    children: React.ReactNode;
+};
+
+export type QueryStateWrapperProps = QueryStateWrapperBaseProps & {
+    queryResult: QueryObserverResult;
+};
+
+export type QueryStatesWrapperProps = QueryStateWrapperBaseProps & {
+    queryResults: QueryObserverResult[];
+    errorCriteria: QueriesDisplayErrorCriteria;
+};
+
+export const QueryStateWrapper: React.FC<QueryStateWrapperProps | QueryStatesWrapperProps> = (
+    props: QueryStateWrapperProps | QueryStatesWrapperProps
+) => {
+    let showQueryLoading = false;
+    let showQueryError = false;
+    if ("queryResult" in props) {
+        showQueryLoading = props.queryResult.isLoading;
+        showQueryError = props.queryResult.isError;
+    } else {
+        showQueryLoading = props.queryResults.some((elm) => elm.isLoading);
+        if (props.queryResults.length > 0) {
```

Comment:
This check to prevent `every` from returning true on empty arrays could be moved to line 41.